src/languageservice/services/yamlCompletion.ts

```diff
@@ -1209,8 +1209,11 @@ export class YamlCompletion {
         return this.getInsertTemplateForValue(value, indent, { index: 1 }, separatorAfter);
       }
     }
-    type = Array.isArray(type) ? type[0] : type;
-    if (type === 'string') {
+    const types = Array.isArray(type) ? type : type;
```

Comment:
What does this line try to achieve. It looks like it is assigning the same value to `types` regardless whether `type` is array or not.

Reply:
Added all the test cases for **number, string, Boolean, null and object**. 

> Also this method called when arrays having ENUM. So I have a question here: Is arrays of ENUM having another array in YAML? if so please provide me a reference schema for that as well. I will update my UT based on that.